src/glum/_glm.py

```diff
@@ -929,7 +929,11 @@ def _make_grid(max_alpha: float) -> np.ndarray:
                     warnings.warn("`min_alpha` is set. Ignoring `min_alpha_ratio`.")
                 min_alpha = self.min_alpha
             return np.logspace(
-                np.log(max_alpha), np.log(min_alpha), self.n_alphas, base=np.e
+                np.log(max_alpha),
+                np.log(min_alpha),
+                self.n_alphas,
+                base=np.e,
+                dtype=X.dtype,
```

Comment:
Added the dtype here. It was causing issues with 32-bit runs because the alphas were set to be 64 bits, and this would upcast downstream variables when multiplying, causing typing issues with our C++ code.